src/com/walmartlabs/lacinia/pedestal/subscriptions.clj

```diff
@@ -480,6 +480,33 @@
    (inject-app-context-interceptor app-context)
    execute-operation-interceptor])
 
+(defn upgrade-request->ring-request
+  [^UpgradeRequest request]
+  (let [headers (.getHeaders request)
+        uri (.getRequestURI request)
+        path (.getPath uri)
+        port (.getPort uri)]
+    (merge {}
```

Comment:
This would probably be more efficient using `cond->` with `assoc` for each optional bit.